src/main/java/uk/gov/ch/developer/docs/controller/errors/DefaultErrorController.java

```diff
@@ -0,0 +1,102 @@
+package uk.gov.ch.developer.docs.controller.errors;
+
+import java.util.HashMap;
+import javax.servlet.RequestDispatcher;
+import javax.servlet.http.HttpServletRequest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.web.servlet.error.ErrorController;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import uk.gov.ch.developer.docs.DocsWebApplication;
+import uk.gov.ch.developer.docs.controller.AbstractPageController;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+@Controller
+@RequestMapping(DefaultErrorController.ERROR_MAPPING)
+
+public class DefaultErrorController extends AbstractPageController implements ErrorController {
+
+
+    private static final Logger LOGGER = LoggerFactory
+            .getLogger(DocsWebApplication.APPLICATION_NAME_SPACE);
+    private static final String TITLE = "Error in Request";
+    public static final String ERROR_MAPPING = "/error";
+
+    @Autowired
+    HttpServletRequest request;
+    @Value("${error.pageNotFound.path}")
+    String notFoundPath;
+
+    @Value("${error.internalService.path}")
+    String serviceError;
+
+    public DefaultErrorController() {
+        super(TITLE);
+    }
+
+    @Override
+    public String getErrorPath() {
+        return ERROR_MAPPING;
+    }
+
+    @Override
+    public String getPath() {
+        return handleError();
+    }
+
+    @Override
+    public boolean isErrorPage() {
+        return true;
+    }
+
+    private String handleError() {
+        // get error status
+        Object status = request.getAttribute(RequestDispatcher.ERROR_STATUS_CODE);
+
+        logError();
+
+        if (status != null) {
```

Comment:
Not a request to change but i'm not a huge fan of nested statements

Reply:
Agreed. 
However this needs to be done before we can safely check statusCode without throwing null pointers and if we extended the nested checks with a not null check that would duplicate code and make it more prone to mistakes if we extend this.